### PR TITLE
impel 1.0.3 (new cask)

### PR DESCRIPTION
**Pull request by ActuallyTaylor**
**Important:** *Do not tick a checkbox if you haven’t performed its action.* Honesty is indispensable for a smooth review process.

_In the following questions `<cask>` is the token of the cask you're submitting._

After making any changes to a cask, existing or new, verify:

- [x] The submission is for [a stable version](https://docs.brew.sh/Acceptable-Casks#stable-versions) or [documented exception](https://docs.brew.sh/Acceptable-Casks#but-there-is-no-stable-version).
- [x] `brew audit --cask --online <cask>` is error-free.
- [x] `brew style --fix <cask>` reports no offenses.

Additionally, **if adding a new cask**:

- [x] Named the cask according to the [token reference](https://docs.brew.sh/Cask-Cookbook#token-reference).
- [x] Checked the cask was not [already refused](https://github.com/search?q=repo%3AHomebrew%2Fhomebrew-cask+is%3Aclosed+is%3Aunmerged+&type=pullrequests) (add your cask's name to the end of the search field).
- [x] Checked the cask is submitted to [the correct repo](https://docs.brew.sh/Acceptable-Casks#finding-a-home-for-your-cask).
- [x] `brew audit --cask --new <cask>` worked successfully.
- [x] `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --cask <cask>` worked successfully.
- [x] `brew uninstall --cask <cask>` worked successfully.

---

### CODE REVIEW

File: Casks/i/impel.rb [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+  url "https://impel-sparkle-updater.fly.dev/dmg/impel_current.dmg", verified: "impel-sparkle-updater.fly.dev/dmg/impel_current.dmg"
+  name "Impel"
+  desc "Proactive AI assistent for the future"
+  homepage "https://www.tryimpel.com/"
+
+  livecheck do
+    url "https://impel-sparkle-updater.fly.dev/appcast.xml"
+    strategy :sparkle do |item|
+      item.version.to_s
+    end
```
Comment (krehel) [inline Casks/i/impel.rb:14]: Can pull the version directly

```suggestion
    strategy :sparkle, &:version
```

Comment (ActuallyTaylor) [reply Casks/i/impel.rb:14]: Yep!

File: Casks/i/impel.rb [hunk 2]
```diff
@@ -0,0 +1,30 @@
+cask "impel" do
+  version "1.0.3"
+  sha256 :no_check
+
+  url "https://impel-sparkle-updater.fly.dev/dmg/impel_current.dmg", verified: "impel-sparkle-updater.fly.dev/dmg/impel_current.dmg"
+  name "Impel"
+  desc "Proactive AI assistent for the future"
```
Comment (krehel) [inline Casks/i/impel.rb:7]: Avoid marketing or empty phrases, please.  Website states "An always-on AI companion for your Mac" so the below should be close enough.

```suggestion
  desc "AI Companion"
```

File: Casks/i/impel.rb [hunk 3]
```diff
@@ -0,0 +1,30 @@
+cask "impel" do
+  version "1.0.3"
+  sha256 :no_check
+
+  url "https://impel-sparkle-updater.fly.dev/dmg/impel_current.dmg", verified: "impel-sparkle-updater.fly.dev/dmg/impel_current.dmg"
```
Comment (krehel) [inline Casks/i/impel.rb:5]: This should be updated to pull from the Sparkle feed, which will provide a versioned download.

File: Casks/i/impel.rb [hunk 4]
```diff
@@ -0,0 +1,30 @@
+cask "impel" do
+  version "1.0.3"
+  sha256 :no_check
```
Comment (krehel) [inline Casks/i/impel.rb:3]: URL can be updated per other comment, then a `sha256` can be placed here.

File: Casks/i/impel.rb [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+  end
+
+  auto_updates true
+  depends_on macos: ">= :sonoma"
+
+  app "Impel.app"
+
+  uninstall quit: "com.tryimpel.impel.mac" # Quit main application
+  uninstall quit: "com.tryimpel.impel.llama-runner-test" # Quit on-device AI application
+
```
Comment (krehel) [inline Casks/i/impel.rb:22]: This should be one single array, no comments.

File: Casks/i/impel.rb (hunk 3 above)
Comment (ActuallyTaylor) [reply Casks/i/impel.rb:5]: Hi! Is the correct syntax for a url that pulls from sparkle `"https://impel-sparkle-updater.fly.dev/dmg/#{version.gsub('.', '_')}.dmg"`?
*We need to replace the . in the version numbers with an underscore for the url.*

Comment (krehel) [reply Casks/i/impel.rb:5]: Should be able to use the enclosure URL in Sparkle, listed as https://impel-sparkle-updater.fly.dev/impel_1_0_3.tar.xz.  We can use a helper method to convert and avoid direct gsub.

url "https://impel-sparkle-updater.fly.dev/impel_#{version.dots_to_underscores}.tar.xz",

Comment (ActuallyTaylor) [reply Casks/i/impel.rb]: Awesome! I just updated the PR

File: Casks/i/impel.rb [hunk 6]
```diff
@@ -0,0 +1,32 @@
+cask "impel" do
+  version "1.0.3"
+  sha256 "627e6e5e4699fab4130fc7a59fbcd6d223317bc6f6e9fe16ad0fe20ab60b690d"
+
+  url "https://impel-sparkle-updater.fly.dev/impel_#{version.dots_to_underscores}.tar.xz", verified: "impel-sparkle-updater.fly.dev/"
```
Comment (krehel) [inline Casks/i/impel.rb:5]: The `verified` should be on its own line, indented to the quote.  Thanks for updating!
```suggestion
  url "https://impel-sparkle-updater.fly.dev/impel_#{version.dots_to_underscores}.tar.xz",
      verified: "impel-sparkle-updater.fly.dev/"
```

**Comment by krehel** [comment]
@ActuallyTaylor - could you help with is stored in `~/Documents/Impel/impel.sqlite`?

Is this something that should be removed if the user completely uninstalls the application with `zap`?

**Comment by ActuallyTaylor** [comment]
> @ActuallyTaylor - could you help with is stored in `~/Documents/Impel/impel.sqlite`?

This is user generated data. I wasn't sure if this should be removed through zap. If it is I can definitely add the whole impel folder there to clear since it's all user generated data.

**Comment by krehel** [comment]
Can you expand on user generated data? And please tell me if I'm wrong in assertions! 😄 

If the data has no residual value after the application is removed, then yes. I'm assuming this is data compiled by the application as part of operation to provide intended features. I would not call this user generated data, maybe user derived or operating data. I'm assuming that's what the `sqlite` file is.

If the folder contains user created artifacts (like reports, documents, etc.) that would be useful after application removal, (like, I fed a PDF into the model and got a summary document back and it's stored here for access) then no we don't want to purge that and wipe out end user artifacts.

**Comment by ActuallyTaylor** [comment]
Yeah I definitely can!
> If the folder contains user created artifacts (like reports, documents, etc.) that would be useful after application removal, (like, I fed a PDF into the model and got a summary document back and it's stored here for access) then no we don't want to purge that and wipe out end user artifacts.

This is basically spot on. All the data in that database are artifacts from the user. Basically any summaries of PDFs, webpages, videos or meetings get stored in that file.

File: Casks/i/impel.rb [hunk 7]
```diff
@@ -0,0 +1,33 @@
...
+
+  url "https://impel-sparkle-updater.fly.dev/impel_#{version.dots_to_underscores}.tar.xz",
+      verified: "impel-sparkle-updater.fly.dev/"
+  name "Impel"
+  desc "AI Companion"
+  homepage "https://www.tryimpel.com/"
+
+  livecheck do
+    url "https://impel-sparkle-updater.fly.dev/appcast.xml"
+    strategy :sparkle, &:version
```
Comment (miccal) [inline Casks/i/impel.rb:13]:
```suggestion
    strategy :sparkle
```

Comment (miccal) [reply Casks/i/impel.rb:13]: Never mind ... move along ... nothing to see here ...

**Comment by krehel** [comment]
Sorry I'm not being difficult but think I'm explaining myself poorly.

There's a difference between that artifact being available for use as the output, and the model using it stored in SQLite. 

Unless I'm missing something I would not expect users to interface with that SQLite db? And that artifacts like output are stored in the same folder.

Meaning we could safely delete that db, but not the directory itself unless empty.

**Comment by ActuallyTaylor** [comment]
I get what your saying, since the SQLite is not a portable format its not something the user can be expected to interface with after the app is gone. Therefore it should be removed.
I will make a commit adding it to the zap list!

**Comment by krehel** [comment]
No worries.  I would personally avoid putting this type of file into the users Documents folder if it is used by the application to function.  It probably should either be placed as hidden resource data in $HOME such as `~/.impel` or into `~/Library/Application Support`

I pushed a commit to fix one last thing, but think this should be good to go now.

Thanks for all your help and working through my nits.

**Comment by miccal** [comment]
Thank you @ActuallyTaylor.

**Review by miccal — APPROVED** [review]

**Pull request merged**